utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/universal/ConfluentUniversalKafkaDeserializerFactory.scala

```diff
@@ -3,30 +3,82 @@ package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.u
 import com.typesafe.scalalogging.LazyLogging
 import io.confluent.kafka.schemaregistry.ParsedSchema
 import io.confluent.kafka.schemaregistry.avro.AvroSchema
+import org.apache.avro.io.DecoderFactory
+import org.apache.kafka.common.header.Headers
 import org.apache.kafka.common.serialization.Deserializer
-import pl.touk.nussknacker.engine.avro.{AvroUtils, RuntimeSchemaData}
-import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.client.ConfluentSchemaRegistryClientFactory
-import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.ConfluentKafkaAvroDeserializer
+import pl.touk.nussknacker.engine.avro.RuntimeSchemaData
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.ConfluentUtils
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.ConfluentUtils.MagicByte
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.client.{ConfluentSchemaRegistryClient, ConfluentSchemaRegistryClientFactory}
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.ConfluentAvroPayloadDeserializer
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.universal.ConfluentUniversalKafkaSerde.SchemaIdHeaderName
 import pl.touk.nussknacker.engine.avro.serialization.KafkaSchemaBasedKeyValueDeserializationSchemaFactory
 import pl.touk.nussknacker.engine.kafka.KafkaConfig
 
+import java.nio.ByteBuffer
 import scala.reflect.ClassTag
+import scala.util.Try
+
+object ConfluentUniversalKafkaSerde {
+  val SchemaIdHeaderName = "schema_id"
+}
+
+class ConfluentUniversalKafkaDeserializer[T](
+                                schemaRegistryClient: ConfluentSchemaRegistryClient,
+                                readerSchemaDataOpt: Option[RuntimeSchemaData[ParsedSchema]],
+                                isKey: Boolean
+                              ) extends Deserializer[T] {
+  private case class WithShiftingRequiredInfo[A](get: A, shouldShiftMagicByte: Boolean) {
```

Comment:
1. get as a field looks strange. maybe `value` instead?
2. am I right that it is not exactly shouldShiftMagicByte but rather shifting header (magic byte + 4 bytes for id)?
3. shouldn't be 0 in case of !shouldShiftMagicByte ?
4. map is unused. looks like value is just a `ParsedSchema`? Maybe the whole that we need is simple tuple?

Reply:
1,2,3 - ok
4 - `map` was used, but now it's not. simpified